src/NuGet.Indexing/NuGetIndexSearcher.cs

```diff
@@ -12,43 +13,59 @@ namespace NuGet.Indexing
 {
     public class NuGetIndexSearcher : IndexSearcher
     {
-        IDictionary<string, Filter> _curatedFeeds;
-        Filter[][] _latest;
+        private readonly IDictionary<string, Filter> _curatedFeeds;
+        private readonly Filter[][] _latest;
 
         public NuGetIndexSearcher(
-            NuGetSearcherManager manager,
+            NuGetSearcherManager manager, 
             IndexReader reader,
-            IndexReader originalReader,
             IDictionary<string, string> commitUserData, 
             IDictionary<string, Filter> curatedFeeds, 
-            Filter[][] latest,
-            VersionsHandler.VersionResult[] versions,
-            IDictionary<string, int> rankings,
-            OpenBitSet latestBitSet,
-            OpenBitSet latestStableBitSet)
+            Filter[][] latest, VersionsHandler.VersionResult[] versions, 
+            RankingsHandler.RankingResult rankings, 
+            OpenBitSet latestBitSet, 
+            OpenBitSet latestStableBitSet,
+            OwnersHandler.OwnersResult owners)
             : base(reader)
         {
             Manager = manager;
-            OriginalReader = originalReader;
             CommitUserData = commitUserData;
-            _curatedFeeds = curatedFeeds;
+
+            _curatedFeeds = new Dictionary<string, Filter>(curatedFeeds.Count);
+            foreach (var curatedFeedsFilter in curatedFeeds)
+            {
+                _curatedFeeds.Add(curatedFeedsFilter.Key, new CachingWrapperFilter(curatedFeedsFilter.Value));
+            }
+            
             _latest = latest;
             Versions = versions;
             Rankings = rankings;
             LatestBitSet = latestBitSet;
             LatestStableBitSet = latestStableBitSet;
+            Owners = owners;
             LastReopen = DateTime.UtcNow;
         }
 
         public NuGetSearcherManager Manager { get; private set; }
-        public IndexReader OriginalReader { get; private set; }
         public IDictionary<string, string> CommitUserData { get; private set; }
         public VersionsHandler.VersionResult[] Versions { get; private set; }
-        public IDictionary<string, int> Rankings { get; private set; }
+        public RankingsHandler.RankingResult Rankings { get; private set; }
         public OpenBitSet LatestBitSet { get; private set; }
         public OpenBitSet LatestStableBitSet { get; private set; }
+        public OwnersHandler.OwnersResult Owners { get; private set; }
         public DateTime LastReopen { get; private set; }
 
+        public Filter GetFilter(string curatedFeed)
+        {
+            Filter curatedFeedFilter;
```

Comment:
Or change it to:
bool TryGetFilter(string curatedFeed, out Filter filter)

---
In reply to: [59766156](https://github.com/NuGet/NuGet.Services.Metadata/pull/64#discussion_r59766156) [](ancestors = 59766156)